apps/meteor/app/cloud/server/functions/syncWorkspace/syncCloudData.ts

```diff
@@ -67,11 +24,17 @@ export async function syncCloudData() {
 
 		const workspaceRegistrationData = await buildWorkspaceRegistrationData(undefined);
 
-		const { license, removeLicense = false } = await fetchWorkspaceSyncPayload({
+		const {
+			license,
+			removeLicense = false,
+			cloudSyncAnnouncement,
+		} = await fetchWorkspaceSyncPayload({
 			token,
 			data: workspaceRegistrationData,
 		});
 
+		await Settings.updateValueById('Cloud_Sync_Announcement_Payload', JSON.stringify(cloudSyncAnnouncement ?? null));
```

Comment:
What happens if cloudSyncAnnouncement is an empty string? Should we save it?

(Also, not sure if you need to notify that this setting has changed so the cached settings get refreshed, can you confirm? 👀 )

Reply:
I'm only saving the setting as string (for now) because we don't currently support objects in a setting such as this one.

This should be changed and improved once we have support for object settings.

Reply:
> Cloud won't send empty strings for this, it is either present with a json payload or not.

I know cloud is ours but can we 100% trust it won't change some payload in the future that could end up breaking this? 👀

Reply:
in this case, we can... the payload will only change if we change the way ui-kit works, so we should be good on this matter.